### ISSUE TITLE
Problem with Geman-McClure loss implementation

### ISSUE THREAD

**Issue by JokerYan** [google/hypernerf#36]
I have been testing the Geman-McClure loss implemented in the repo, more specifically the model_utils.general_loss_with_squared_residual() with alpha = -2.

However, the output, as shown below, is very different from my expectation. (alpha = -2, scale = 0.1)
![image](https://user-images.githubusercontent.com/18729314/184272219-a7c3d744-351d-480f-83b5-a892b107e0ec.png)
Shouldn't it be like the quadratic shape when |x| < scale

**Comment by keunhong** [comment]
Did you square the residual before passing it to the function?

**Comment by JokerYan** [comment]
No, I didn't. I think that is the problem. Thank you very much!
